opm/input/eclipse/EclipseState/Grid/FieldProps.cpp

```diff
@@ -672,16 +672,64 @@ bool FieldProps::supported<int>(const std::string& keyword) {
 }
 
+void FieldProps::apply_multipliers()
+{
+    static const auto prefix = FieldPropsManager::getMultiplierPrefix();
+
+    for(auto pos =multiplier_kw_infos_.begin(); pos != multiplier_kw_infos_.end(); ++pos)
+    {
+        const auto& [mult_keyword, kw_info] = *pos;
+        const std::string keyword = mult_keyword.substr(prefix.size());
+        auto mult_iter = this->double_data.find(mult_keyword);
+        assert(mult_iter != this->double_data.end());
+        auto iter = this->double_data.find(keyword);
+        if (iter == this->double_data.end()) {
+            iter = this->double_data
+                .emplace(std::piecewise_construct,
+                         std::forward_as_tuple(keyword),
+                         std::forward_as_tuple(kw_info, this->active_size, kw_info.global ? this->global_size : 0))
+                .first;
+        }
+        auto multiplier = mult_iter->second.data.begin();
+        for(auto data = iter->second.data.begin(); data != iter->second.data.end(); ++data, ++multiplier)
+        {
+            *data *= *multiplier;
+        }
+        // If data is global, then we also need to set the global_data. I think they should be the same at this stage, though!
+        if (kw_info.global)
+        {
+            assert(mult_iter->second.global_data.has_value() && iter->second.global_data.has_value());
+            multiplier = mult_iter->second.global_data.value().begin();
+            auto& global_array = iter->second.global_data.value();
+            for(auto data = global_array.begin(); data != global_array.end(); ++data, ++multiplier)
+            {
+                *data *= *multiplier;
+            }
+        }
+        this->double_data.erase(mult_iter);
+    }
+    multiplier_kw_infos_.clear();
+}
 
 template <>
-Fieldprops::FieldData<double>& FieldProps::init_get(const std::string& keyword_name, const Fieldprops::keywords::keyword_info<double>& kw_info) {
+Fieldprops::FieldData<double>& FieldProps::init_get(const std::string& keyword_name, const Fieldprops::keywords::keyword_info<double>& kw_info,
+                                                    bool multiply_variant) {
+    
+    if(multiply_variant && !kw_info.scalar_init.has_value())
+        OPM_THROW(std::logic_error, std::string("Keyword " +  keyword_name + " is a multiplier and should have a default initial value."));
     const std::string& keyword = Fieldprops::keywords::get_keyword_from_alias(keyword_name);
+    const std::string& mult_keyword = std::string(multiply_variant? FieldPropsManager::getMultiplierPrefix() : "") + keyword;
 
-    auto iter = this->double_data.find(keyword);
-    if (iter != this->double_data.end())
+    auto iter = this->double_data.find(mult_keyword);
+    if (iter != this->double_data.end()) {
         return iter->second;
+    } else if(multiply_variant){
+        assert(keyword != ParserKeywords::PORV::keywordName || keyword != ParserKeywords::TEMPI::keywordName ||
```

Comment:
What does this try to express?  Suppose the keyword is `PORV`.  Then `keyword != PORV` is false, but `keyword != TEMPI` is true and the overall condition would still be true.  Is that intentional?